tests/integration/tket_optimization/test_optimization.py

```diff
@@ -0,0 +1,66 @@
+from guppylang import guppy
+from guppylang.std.quantum import qubit, cx, h, s, t
+
+from pytket.passes import RemoveRedundancies, CliffordSimp
+
+from tket.passes import NormalizeGuppy, PytketHugrPass, PassResult
+
+from hugr.hugr.base import Hugr
+
+
+def _count_ops(hugr: Hugr, string_name: str) -> int:
+    count = 0
+    for _, data in hugr.nodes():
+        if string_name in data.op.name():
+            count += 1
+
+    return count
+
+
+normalize = NormalizeGuppy()
+
+
+def test_guppy_normalization() -> None:
+    @guppy
+    def pauli_zz_rotation(q0: qubit, q1: qubit) -> None:
+        cx(q0, q1)
+        t(q1)
+        cx(q0, q1)
+
+        normalized_hugr: Hugr = normalize(
+            pauli_zz_rotation.compile_function().modules[0]
+        )
+
+        # Test that the dataflow block is inlined by NormalizeGuppy
+        assert _count_ops(normalized_hugr, "DataflowBlock") == 0
```

Comment:
Maybe for explicitness I should assert that `_count_ops(hugr, "DataflowBlock") == 1` prior to the normalization pass so its clear to the reader that the test does something.